pkg/asset-manager-utils/test/SinglePoolAaveATokenAssetManager.test.ts

```diff
@@ -0,0 +1,330 @@
+        const amountToWithdraw = (await assetManager.maxInvestableBalance(poolId)).mul(-1);
+        // await assetManager.connect(poolController).setInvestablePercent(poolId, fp(0));
+
+        await expectBalanceChange(() => assetManager.connect(lp).capitalOut(poolId, amountToWithdraw), tokens, [
+          { account: lendingPool.address, changes: { DAI: ['near', -amountToWithdraw] } },
+          { account: vault.address, changes: { DAI: ['near', amountToWithdraw] } },
+        ]);
+      });
+
+      it("updates the pool's managed balance", async () => {
+        const maxInvestableBalance = await assetManager.maxInvestableBalance(poolId);
+
+        // return a portion of the return to the vault to serve as a buffer
+        const amountToWithdraw = maxInvestableBalance.abs();
+
+        await assetManager.connect(lp).capitalOut(poolId, amountToWithdraw);
+
+        const { managed } = await vault.getPoolTokenInfo(poolId, tokens.DAI.address);
+        const actualManagedBalance = await assetManager.readAUM();
+
+        expect(managed).to.be.eq(actualManagedBalance);
+      });
+
+      it('allows the pool to withdraw tokens to rebalance', async () => {
+        const maxInvestableBalance = await assetManager.maxInvestableBalance(poolId);
+
+        // return a portion of the return to the vault to serve as a buffer
+        const amountToWithdraw = maxInvestableBalance.abs();
+
+        await expectBalanceChange(() => assetManager.connect(lp).capitalOut(poolId, amountToWithdraw), tokens, [
+          { account: lendingPool.address, changes: { DAI: ['near', -amountToWithdraw] } },
+          { account: vault.address, changes: { DAI: ['near', amountToWithdraw] } },
+        ]);
+      });
+    });
+  });
+
+  describe('ClaimRewards', () => {
+    const rewardAmount = fp(1);
+
+    beforeEach(async () => {
+      const bptBalance = await pool.balanceOf(lp.address);
+      await pool.connect(lp).approve(stakingContract.address, bptBalance);
+      await stakingContract.connect(lp)['stake(uint256)'](bptBalance.mul(3).div(4));
+
+      // Stake half of the BPT to another address
+      await stakingContract.connect(lp)['stake(uint256,address)'](bptBalance.div(4), other.address);
+    });
+
+    it('sends expected amount of stkAave to the rewards contract', async () => {
+      const rewardsBefore = await stkAave.balanceOf(stakingContract.address);
+      await assetManager.claimRewards();
+      const rewardsAfter = await stkAave.balanceOf(stakingContract.address);
+      expect(rewardsAfter).to.be.eq(rewardsBefore.add(rewardAmount));
+    });
+
+    it('distributes the reward according to the fraction of staked LP tokens', async () => {
+      await assetManager.claimRewards();
+      await advanceTime(10);
+
+      const expectedReward = fp(0.75);
+      const actualReward = await stakingContract.earned(lp.address, stkAave.address);
+      expect(expectedReward.sub(actualReward).abs()).to.be.lte(100);
+    });
+  });
+});
```

Comment:
We seem to be missing tests for `updateStakingContract()` and `readAUM()` here. I believe the goal is to rework the notion of AUM, cache and so on? If so, maybe it doesn't make sense to spend time working on these now.

Reply:
removed `updateStakingContract` for now - I think it's likely that we would be doing an upgrade from a merkle drop to a staking contract, if anything (hoping we can just start use MultiMultiRewards from the beginning)